shell-integration/src/streaming_state.rs

```diff
@@ -0,0 +1,30 @@
+// Copyright (c) SimpleStaking, Viable Systems and Tezedge Contributors
+// SPDX-License-Identifier: MIT
+
+use std::collections::HashMap;
+
+use futures::task::Waker;
+use uuid::Uuid;
+
+pub type StreamWakers = HashMap<Uuid, Waker>;
+
+pub trait StreamCounter {
+    fn get_mutable_streams(&mut self) -> &mut StreamWakers;
+    fn get_streams(&self) -> StreamWakers;
```

Comment:
@adonagy 
if you return `& StreamWakers`, does it work?
I guess, that cloning is unnecessery

here the iter takes just reference, right?

```
   self.get_streams().iter().for_each(|(_, waker)| waker.wake_by_ref())
```

Reply:
Yep, that's a very good point :) Adjusted accordingly